sdk/src/androidTest/java/com/mongodb/stitch/StitchClientTest.kt

```diff
@@ -0,0 +1,218 @@
+package com.mongodb.stitch
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.support.test.InstrumentationRegistry
+import android.support.test.runner.AndroidJUnit4
+import com.mongodb.stitch.android.AuthListener
+import com.mongodb.stitch.android.PipelineStage
+import com.mongodb.stitch.android.StitchClient
+import com.mongodb.stitch.android.auth.anonymous.AnonymousAuthProvider
+import com.mongodb.stitch.android.auth.oauth2.facebook.FacebookAuthProviderInfo
+import com.mongodb.stitch.android.auth.oauth2.google.GoogleAuthProviderInfo
+import io.appflate.restmock.RESTMockServer
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import kotlin.reflect.full.declaredMemberProperties
+import kotlin.reflect.jvm.isAccessible
+import io.appflate.restmock.utils.RequestMatchers.pathContains
+import org.json.JSONObject
+import io.appflate.restmock.android.AndroidLogger
+import io.appflate.restmock.android.AndroidAssetsFileParser
+import io.appflate.restmock.RESTMockServerStarter
+import io.appflate.restmock.utils.RequestMatchers.pathEndsWith
+
+/**
+ * Test public methods of StitchClient, running through authentication
+ * and pipeline flows.
+ */
+@RunWith(AndroidJUnit4::class)
+class StitchClientTest {
+    companion object {
+        const val FakeAccessToken = "fake-access-token"
```

Comment:
Looks like I was still thinking Swift-y here. Should be defaulting to Java naming conventions here: http://kotlinlang.org/docs/reference/coding-conventions.html will change.